### PR TITLE
planner: fix the problem of not finding the referenced column in the virtual column

### LINKED ISSUE #57079: analyze cannot be with json's virtual column
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
CREATE TABLE `t1` (
  `id` bigint(20) NOT NULL /*T![auto_rand] AUTO_RANDOM(5) */ ,
  `c1` json DEFAULT NULL ,
  `c2` varchar(255) GENERATED ALWAYS AS (json_unquote(json_extract(`c1`, '$.code'))) VIRTUAL ,
  PRIMARY KEY (`id`) /*T![clustered_index] CLUSTERED */
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin /*T![auto_rand_base] AUTO_RANDOM_BASE=7765648957 */ ;
analyze table t1
```

### 2. What did you expect to see? (Required)

succeed to run 
### 3. What did you see instead (Required)

```
(1054, "Unknown column 'c1' in 'expression'")
```

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #57079

Problem Summary:

### What changed and how does it work?

the problem happen at the ```executorBuilder.buildAnalyze```.

```ColumnInfos2ColumnsAndNames``` will find the virtual reference column by the ```column Infos```. but when we enable ```analyze_skip_column_types```, ```column Infos``` will lost some columns. so it cannot find its needed column.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/builder.go [hunk 1]
```diff
@@ -3061,13 +3061,23 @@ func (b *executorBuilder) buildAnalyze(v *plannercore.Analyze) exec.Executor {
...
 	exprCtx := b.ctx.GetExprCtx()
 	for _, task := range v.ColTasks {
+		// ColumnInfos2ColumnsAndNames will use the `colInfos` to find the unique id for the column,
+		// so we need to make sure all the columns pass into it.
 		columns, _, err := expression.ColumnInfos2ColumnsAndNames(
 			exprCtx,
 			pmodel.NewCIStr(task.AnalyzeInfo.DBName),
 			task.TblInfo.Name,
-			task.ColsInfo,
+			append(task.ColsInfo, task.SkipColsInfo...),
```
Comment (AilinKid) [inline pkg/executor/builder.go:3070]: if we don't merge them here, this func couldn't find the unique id and return err?

Comment (hawkingrei) [reply pkg/executor/builder.go:3070]: Yes, it cannot find this unique id and return err.

**Review by AilinKid — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Review by windtalker — APPROVED** [review]
lgtm

**Pull request merged**